### PR TITLE
Feature/isolate gru

### PR DESCRIPTION
**Pull request by martynpclark**
Refactor to isolate HRU computations. Tested before HRU refactor with identical results.

### CODE REVIEW

File: build/source/engine/run_oneGRU.f90 [hunk 1]
```diff
@@ -79,141 +68,142 @@ module run_oneGRU_module
  ! simulation for a single GRU
  subroutine run_oneGRU(&
                        ! model control
-                       iGRU,               & ! intent(in):    GRU index
```
Comment (bartnijssen) [inline build/source/engine/run_oneGRU.f90]: Do you want to pass the `iGRU` still into `run_oneGRU` simply for logging and error reporting purposes. In RASM we have had the problem in earlier versions that we would have an error in a low level routine and we wouldn't know where the error happened. If we get a complete stack trace when we report the error then that is not a problem. (note that it is a good sign that you do not need this info AT ALL to run the routine)

Comment (bartnijssen) [reply build/source/engine/run_oneGRU.f90]: Or is this part of `gruInfo`?

**Comment by bartnijssen** [comment]
This series of PRs is really excellent for moving SUMMA forward. I know it makes no difference in the result, but it will enable much better scaling down the road!

File: build/source/engine/run_oneGRU.f90 (hunk 1 above)
Comment (martynpclark) [reply build/source/engine/run_oneGRU.f90]: We can get a traceback without including iGRU

Comment (bartnijssen) [reply build/source/engine/run_oneGRU.f90]: Perfect!

Comment (martynpclark) [reply build/source/engine/run_oneGRU.f90]: The routine will exit gracefully, and then IGRU will be available.

**Pull request merged**